dcicutils/redis_utils.py

```diff
@@ -0,0 +1,89 @@
+import redis
+# Low level utilities for working with Redis
+
+
+def create_redis_client(*, url, ping=True):
+    """ Creates a Redis client connecting from a host:port or a URL
+        Because this creation pings Redis should re-use this as much as possible, but
+        you can skip the ping if you're confident it will come up
+    """
+    r = redis.from_url(url)
+    if ping:
+        r.ping()
+    return r
+
+
+class RedisException(Exception):
+    pass
+
+
+class RedisBase(object):
+    """ This class contains low level methods meant to implement useful Redis APIs. The idea is these functions
+        are used to implement the methods needed in Redis Tools.
+
+        One important note about Redis - the database wants bytes like objects in the
+        place of strings, so some automatic encoding/decoding from utf-8 is implemented
+        here and thus should NOT be done by the caller. If the caller does do it though
+        it is assumed it has been done correctly, or is of a type (ie: float) that does
+        not require encoding/decoding.
+
+        TODO: implement more APIs as needed - Redis is very complex and offers a lot of
+              potentially useful functionality, and it may be necessary to split off into
+              sub classes later on to isolate in a reasonable manner, but for now all here
+    """
+
+    def __init__(self, redis_handle):
+        self.redis = redis_handle
+
+    @staticmethod
+    def _encode_value(value):
+        """ Function used to preprocess all values that are strings and encode them as utf-8 bytes """
+        return value.encode('utf-8') if (value and isinstance(value, str)) else value
```

Comment:
So this is not needed for other types? Or those other types are not given? Something about the expected args would be helpful in the doc string. I was assuming you'd just `json.dumps(value).encode('utf-8')` to do the encoding and `json.loads(value.decode('utf-8'))` to pull the value back, but maybe you're trying to be more efficient than that. If so, should you perhaps error check the argument type, or, minimally, declare the argument type you expect (which has only informational and not runtime effect)?